workerPool.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+)
+
+//worker blocks until the object ids are available, and sends gotten details to result
+func (req *client) worker(ctx context.Context, jobs <-chan []int, seen map[int]struct{}, result chan<- ObjectDetail) {
+	for {
+		for ids := range jobs {
+			for _, objectID := range ids {
+				req.RLock()
+				_, ok := seen[objectID]
+				req.RUnlock()
+				if ok {
+					continue
+				}
+
+				req.Lock()
+				seen[objectID] = struct{}{}
+				req.Unlock()
+
+				go func(id int) {
```

Comment:
Not quite what to expected. You should preallocate some amount of go funcs with workers.
```
for i :=0; i < workerCount; i++ {
    go runWorker(ctx, <chan_with_jobs>, <chan_with_results>)
}
```
in every `runWorker` func you need to have infinite loop who will iterate over your jobs chan and return result in result chan (or something similar).